### PR TITLE
9 iso sfn

### PR DESCRIPTION
**Pull request by Robinlovelace**
- Add working code for #9, feedback welcome
- Show results

### CODE REVIEW

File: README_files/figure-gfm/outputs-1.png
Comment (Robinlovelace) [inline README_files/figure-gfm/outputs-1.png:1]: This isn't looking right to me @bda1986, heads-up that there may be issues with the code in `iso_sfn()` function, your comments v. welcome.

**Review by Robinlovelace — COMMENTED** [review]
Not yet ready...

File: README_files/figure-gfm/outputs-1.png
Comment (bda1986) [reply README_files/figure-gfm/outputs-1.png:1]: Indeed looking a bit strange. I'll take a closer look and cross-compare it with the results based on OSRM.

Comment (bda1986) [reply README_files/figure-gfm/outputs-1.png:1]: I'm still trying to figure this one out, Robin.
It is weird that the cells in which the points are located are able to change ownership.
That should not be possible, as is correctly shown for OSRM based isonois.

From left to right: `iso_vnoi()`, `iso_sfnoi()` and `iso_osrm()`
grafik

I've tried a higher resolution (10) and the results look even stranger.

From left to right: `iso_vnoi()`, `iso_sfnoi()` and `iso_osrm()`
grafik

My guess is that the routing based on `sfnetworks` is not producing the correct outcomes.
I have to familiarize myself with that though, again sorry for the delay!

Comment (Robinlovelace) [reply README_files/figure-gfm/outputs-1.png:1]: Agreed, something really not right here! Thanks for testing Bernhard. My guess: there's a bug in `iso_sfn()`..

**Comment by Robinlovelace** [comment]
High resolution results for `iso_osrm()` looking great btw.

File: README_files/figure-gfm/outputs-1.png
Comment (bda1986) [reply README_files/figure-gfm/outputs-1.png:1]: ... I seem to have found and fixed the bug: da83eb9d3e3b8a9fcb54617982b84f193a3a4600

When selecting the minimal cost path, the index of the paths did not correspond to the index/position of the points in the original data set. It's looking much smoother and less erratic now.

File: R/iso_grid.R [hunk 1]
```diff
@@ -329,15 +329,15 @@ iso_sfn = function(points, net, grid, n = 3) {
...
+      net |>
+        sfnetworks::activate("edges") |>
         dplyr::slice(paths$edge_paths[[j]]) |>
         dplyr::mutate(route_number = j) |>
         sf::st_as_sf() |>
         dplyr::summarise(length = sum(weight)) |>
         dplyr::pull(length)
     })
-    point_shortest_id = which.min(path_weights)
+    point_shortest_id = nearest_point_ids[which.min(path_weights)]
```
Comment (Robinlovelace) [inline R/iso_grid.R:340]: Looks good!

**Review by Robinlovelace — COMMENTED** [review]
Great work, any ideas what was causing the issue?

**Review by bda1986 — APPROVED** [review]

**Pull request merged**